### DISCUSSION TITLE
Node.js errors

### DISCUSSION THREAD

**Discussion by Kobierecky** [thedevs-network/kutt#516] [Q&A]
Trying to install and getting errors, node.js is installed

[23:10] [server1.x.com cutitto] # npm run dev

> kutt@2.7.1 dev /home/x/cutit/public/cutitto
> npm run migrate && nodemon server/server.ts

> kutt@2.7.1 migrate /home/x/cutit/public/cutitto
> knex migrate:latest --env production

sh: knex: command not found

npm ERR! Linux 3.10.0-1127.19.1.el7.x86_64
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "migrate"
npm ERR! node v6.17.1
npm ERR! npm  v3.10.10
npm ERR! file sh
npm ERR! code ELIFECYCLE
npm ERR! errno ENOENT
npm ERR! syscall spawn
npm ERR! kutt@2.7.1 migrate: `knex migrate:latest --env production`
npm ERR! spawn ENOENT
npm ERR!
npm ERR! Failed at the kutt@2.7.1 migrate script 'knex migrate:latest --env production'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the kutt package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     knex migrate:latest --env production
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs kutt
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls kutt
npm ERR! There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! Please include the following file with any support request:
npm ERR!     /home/x/cutit/public/cutitto/npm-debug.log

npm ERR! Linux 3.10.0-1127.19.1.el7.x86_64
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "run" "dev"
npm ERR! node v6.17.1
npm ERR! npm  v3.10.10
npm ERR! code ELIFECYCLE
npm ERR! kutt@2.7.1 dev: `npm run migrate && nodemon server/server.ts`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the kutt@2.7.1 dev script 'npm run migrate && nodemon server/server.ts'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the kutt package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run migrate && nodemon server/server.ts
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs kutt
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls kutt
npm ERR! There is likely additional logging output above.
npm WARN Local package.json exists, but node_modules missing, did you mean to install?

npm ERR! Please include the following file with any support request:
npm ERR!     /home/x/cutit/public/cutitto/npm-debug.log

**Comment by poeti8** [comment]
Did you run `npm install` to install dependencies first?

**Reply by Kobierecky** [reply]
Yes, of course. What version of node.js is required?

**Reply by poeti8** [reply]
Could be an access issue where it can't run scripts from `node_modules/.bin`.

**Reply by Kobierecky** [reply]
Was running on root
